wikiHandler.js

```diff
@@ -86,10 +93,46 @@ function formatPrice(price) {
     }
 }
 
+function parsePage(page, server) {
+    const $ = cheerio.load(page);
+    const auc_data = $(`#auc_${server} .eoTable3 td`).contents();
+    let price_data = {};
+    if (auc_data !== undefined && auc_data[0] !== undefined) {
+        price_data[30] = auc_data[0].data.trim();
+    }
+    if (auc_data !== undefined && auc_data[1] !== undefined) {
+        price_data[90] = auc_data[1].data.trim();
+    }
+    return price_data;
+}
+
 async function getWikiPricing(item_url, server) {
     // Translate server name to "capitalized" form, eg: GREEN -> Green
     server = server[0] + server.slice(1).toLowerCase();
 
+    //check cache before fetching
+    cache.get(item_url, (err, cached_data) => {
+        if (err) console.error(err);
+        //use cached data if available
+        if (cached_data) {
+            console.log('cached worked for ', item_url)
+            parsePage(cached_data, server)
```

Comment:
Second comment here -- after we do the parsePage, does it ... not return the parsed data? Do we fall through to like 136 below and run fetch() a second time?

Reply:
Oof. Yeah, my bad- remnants of your code i was pulling apart.  Deleted now.